src/ui/hook/useTimeout.ts

```diff
@@ -0,0 +1,23 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+type ClearTimeout = () => void
+type UseTimeout = { clear: ClearTimeout }
+
+export const useTimeout = (callback: () => void, delay: number | null): UseTimeout => {
```

Comment:
I don't have time to study the finesse of the implementation but I feel that it is possible to simplify it into just 2 effects, one for the callback, another one for the timeout itself, returning the disposition directly.

Reply:
Ok. As seen with you @ErikssonJoakim, it doesn't seem obvious that the function can be improved. So I propose to keep the current approach which seems to me good. 😌